py/selenium/webdriver/safari/webdriver.py

```diff
@@ -46,6 +47,7 @@ def __init__(self, port=0, executable_path="/usr/bin/safaridriver", reuse_servic
 
         self._reuse_service = reuse_service
         if not reuse_service:
+            self.service = Service(executable_path, port=port, quiet=quiet)
```

Comment:
This needs to be assigned outside the branch because we always want to have a service instance.
self.service is used unconditionally when calling the super initializer, and other code expects it to be there even if reuse_service is True.